src/shared/utils/flattenChildren.js

```diff
@@ -46,7 +52,7 @@ function flattenSingleChildIntoContext(traverseContext, child, name, selfDebugID
  * children will not be included in the resulting object.
  * @return {!object} flattened children keyed by name.
  */
-function flattenChildren(children, selfDebugID) {
+function flattenChildren(children: ReactElement<any>, selfDebugID: number): any {
```

Comment:
Can you type the return function, if we put any, it doesn't bring any benefit since flow is just going to ignore it.

```js
?{ [name: string}: ReactElement<any> }
```